airflow/models/xcom.py

```diff
@@ -499,6 +509,27 @@ def orm_deserialize_value(self) -> Any:
         return BaseXCom.deserialize_value(self)
 
 
+def serialize_value_shim(**kwargs):
+    """
+    Previously XCom.serialize_value only accepted one argument ``value``.  In order to give
+    custom XCom backends more flexibility with how they store values we now forward to
+    ``XCom.serialize_value`` all params passed to ``XCom.set``.  In order to maintain
+    compatibility with XCom backends written with the old signature we need to use this
+    compatibility shim, which forwards only those kwargs which are accepted by the
+    XCom backend.
+    """
+    signature = inspect_function_arguments(XCom.serialize_value)
+    kwargs = {k: kwargs.get(k) for k in signature.bound_arguments}
```

Comment:
I think we should do this in resolve_xcom_backend, rather than on every call to this.

Reply:
done